### PR TITLE
60: Add Image and InlineImages components

### LINKED ISSUE #60: Add Image and InlineImages components
Subtypes:
- Border
- Shadow

### PR DESCRIPTION
**Pull request by Caleb-Ellis**
# Done
- Added [Image](https://docs.vanillaframework.io/en/patterns/images) and [InlineImages](https://docs.vanillaframework.io/en/patterns/inline-images) component
- Includes Default, Shadowed, Bordered, Bordered + Shadowed subtypes for Image, and Default for InlineImages
- Added stories to Storybook
- Added snapshot tests

# QA
- Pull code
- Run `yarn clean` and `yarn serve` and navigate to http://localhost:6006/
- Run `yarn lint` and `yarn test` to ensure there are no linting or testing errors
- Verify the Image and InlineImages components in Storybook match the Vanilla docs

# Fixes
Fixes #60

### CODE REVIEW

File: src/lib/components/Image/Image.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+};
+
+Image.defaultProps = {
+  bordered: false,
+  shadowed: false,
+};
+
+Image.propTypes = {
+  src: PropTypes.string.isRequired,
+  alt: PropTypes.string.isRequired,
```
Comment (anthonydillon) [inline src/lib/components/Image/Image.js]: I dont think the `alt` should be required by just default to an empty string.

File: src/lib/components/Image/Image.test.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    expect(json).toMatchSnapshot();
+  });
+
+  it('should render Image with props correctly', () => {
+    const image = ReactTestRenderer.create(
+      <div>
+        <Image src="http://placekitten.com/g/300/300" alt="" />
+        <Image bordered src="http://placekitten.com/g/300/300" alt="" />
+        <Image shadowed src="http://placekitten.com/g/300/300" alt="" />
+        <Image bordered shadowed src="http://placekitten.com/g/300/300" alt="" />
```
Comment (anthonydillon) [inline src/lib/components/Image/Image.test.js]: It would be better to split these into separate tests so its clear what is broken when tests fail.

**Review by anthonydillon — CHANGES_REQUESTED** [review]
Few suggestions.

File: src/lib/components/Image/Image.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import './Image.css';
+
+const Image = (props) => {
+  let classArray = [];
+
+  if (props.bordered) {
+    classArray = classArray.concat('p-image--bordered');
```
Comment (tbille) [inline src/lib/components/Image/Image.js]: use `.push` instead, `.concat` is to [merge arrays](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/concat).  We should keep the meaning of the functions even if they give the same result

File: src/lib/components/Image/Image.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  }
+  if (props.shadowed) {
+    classArray = classArray.concat('p-image--shadowed');
```
Comment (tbille) [inline src/lib/components/Image/Image.js]: same here

File: src/lib/components/Image/Image.js (hunk 3 above)
Comment (Caleb-Ellis) [reply src/lib/components/Image/Image.js]: Using `.concat` was a deliberate choice to not mutate the original array, which isn't an issue here but treating arrays as immutable is useful when dealing with state changes and lifecycle events in React (and by extension Redux). You've reminded me about the spread operator though, which achieves the same thing but in a cleaner syntax (in my opinion).

**Review by tbille — APPROVED** [review]

**Review by anthonydillon — APPROVED** [review]
Thanks :+1:

**Pull request merged**